builder/table_test.go

```diff
@@ -133,3 +133,82 @@ func TestTable_Build(t *testing.T) {
 		})
 	}
 }
+
+func TestTable_BuildWithDefinitionFilter(t *testing.T) {
+	var (
+		definitionFilter = func(table rel.Table, def rel.TableDefinition) bool {
+			_, ok := def.(rel.Key)
+			// https://www.sqlite.org/omitted.html
+			// > Only the RENAME TABLE, ADD COLUMN, RENAME COLUMN, and DROP COLUMN variants of the ALTER TABLE command are supported.
+			if ok && table.Op == rel.SchemaAlter {
+				return false
+			}
+
+			return true
+		}
+		tableBuilder = Table{
+			BufferFactory:    BufferFactory{InlineValues: true, BoolTrueValue: "true", BoolFalseValue: "false", Quoter: Quote{IDPrefix: "`", IDSuffix: "`", IDSuffixEscapeChar: "`", ValueQuote: "'", ValueQuoteEscapeChar: "'"}},
+			ColumnMapper:     sql.ColumnMapper,
+			DefinitionFilter: definitionFilter,
+		}
+	)
+
+	tests := []struct {
+		result string
+		table  rel.Table
+	}{
+		{
+			result: "CREATE TABLE `columns` (`bool` BOOL NOT NULL DEFAULT false, `int` INT(11) UNSIGNED, `bigint` BIGINT(20) UNSIGNED, `float` FLOAT(24) UNSIGNED, `decimal` DECIMAL(6,2) UNSIGNED, `string` VARCHAR(144) UNIQUE, `text` TEXT(1000), `date` DATE, `datetime` DATETIME DEFAULT '2020-01-01 01:00:00', `time` TIME, `blob` blob, PRIMARY KEY (`int`), FOREIGN KEY (`int`, `string`) REFERENCES `products` (`id`, `name`) ON DELETE CASCADE ON UPDATE CASCADE, UNIQUE `date_unique` (`date`)) Engine=InnoDB;",
+			table: rel.Table{
+				Op:   rel.SchemaCreate,
+				Name: "columns",
+				Definitions: []rel.TableDefinition{
+					rel.Column{Name: "bool", Type: rel.Bool, Required: true, Default: false},
+					rel.Column{Name: "int", Type: rel.Int, Limit: 11, Unsigned: true},
+					rel.Column{Name: "bigint", Type: rel.BigInt, Limit: 20, Unsigned: true},
+					rel.Column{Name: "float", Type: rel.Float, Precision: 24, Unsigned: true},
+					rel.Column{Name: "decimal", Type: rel.Decimal, Precision: 6, Scale: 2, Unsigned: true},
+					rel.Column{Name: "string", Type: rel.String, Limit: 144, Unique: true},
+					rel.Column{Name: "text", Type: rel.Text, Limit: 1000},
+					rel.Column{Name: "date", Type: rel.Date},
+					rel.Column{Name: "datetime", Type: rel.DateTime, Default: time.Date(2020, 1, 1, 1, 0, 0, 0, time.UTC)},
+					rel.Column{Name: "time", Type: rel.Time},
+					rel.Column{Name: "blob", Type: "blob"},
+					rel.Key{Columns: []string{"int"}, Type: rel.PrimaryKey},
+					rel.Key{Columns: []string{"int", "string"}, Type: rel.ForeignKey, Reference: rel.ForeignKeyReference{Table: "products", Columns: []string{"id", "name"}, OnDelete: "CASCADE", OnUpdate: "CASCADE"}},
+					rel.Key{Columns: []string{"date"}, Name: "date_unique", Type: rel.UniqueKey},
+				},
+				Options: "Engine=InnoDB",
+			},
+		},
+		{
+			result: "ALTER TABLE `columns` ADD COLUMN `verified` BOOL;ALTER TABLE `columns` RENAME COLUMN `string` TO `name`;ALTER TABLE `columns` ;ALTER TABLE `columns` DROP COLUMN `blob`;",
+			table: rel.Table{
+				Op:   rel.SchemaAlter,
+				Name: "columns",
+				Definitions: []rel.TableDefinition{
```

Comment:
should we just filter out silently like this? or do you think it's better to print some log? 🤔

Reply:
It may be sufficient to output logs in the `DefinitionFilter`

Reply:
agree, something like that should be sufficient

Reply:
Hi @youpy, could you add the log as well?

Reply:
Sorry, I just added.

Reply:
I thought the log will be inside definitions functions?

Reply:
https://github.com/go-rel/sql/pull/39/files#diff-70314c95aab2c79824512084ba4bd3799c061dcf5ebad5e3695b6486462261f2R246

Reply:
Oh, I misunderstood
added ([53e002a](https://github.com/go-rel/sql/pull/39/commits/53e002a470966576e913d592d9b8340e65a8f143))